### ISSUE TITLE
I'm not able to get Apprise working

### ISSUE THREAD

**Issue by tmknight** [tmknight/docker-autoheal#107]
I'm not able to get Apprise working. I get no notifications...

The logs look like something should appear:

```
2024-08-06T01:37:16.752741576Z 2024-08-06 01:37:16+0000 [WARNING] [OnlyOfficeDocumentServer (745494b32ac0)] Container is unhealthy with 3 failures
2024-08-06T01:37:16.752798386Z 2024-08-06 01:37:16+0000 [WARNING] [OnlyOfficeDocumentServer (745494b32ac0)] Container last output: [1] https://xyz.de/healthcheck:
2024-08-06T01:37:16.752805556Z Remote file does not exist -- broken link!!!
2024-08-06T01:37:16.752812627Z 
2024-08-06T01:37:16.752817842Z 2024-08-06 01:37:16+0000 [WARNING] [OnlyOfficeDocumentServer (745494b32ac0)] Container restarting with 30s timeout
2024-08-06T01:37:47.022011919Z 2024-08-06 01:37:47+0000 [   INFO] [OnlyOfficeDocumentServer (745494b32ac0)] Container restart was successful
2024-08-06T01:37:47.291852731Z 2024-08-06 01:37:47+0000 [   INFO] Response: 401 Unauthorized
```

I guess I misunderstood it. You do not include Apprise right? So I need the webhook URL option but it does also not work with Discord. Any working example for Discord / Telegram?

_Originally posted by @saschabrockel in https://github.com/tmknight/docker-autoheal/issues/106#issuecomment-2270202142_

**Comment by tmknight** [comment]
The logs are telling the story and seem to indicate the remote endpoint is reached and responds with permission issue.  You will have to troubleshoot on your side the correct format of the URL.

**Comment by saschabrockel** [comment]
The readme is confusing me. Olivetin for example does not include apprise and you need a separate instance. Is it the same case here? I know Apprise and also the format and tried literally all of them and it is not working. That's why I'm so confused. I mean the syntax is clear: https://github.com/caronc/apprise/wiki/Notify_discord

Lately, I tried `AUTOHEAL_WEBHOOK_URL` with https://discordapp.com/api/webhooks/120131117899.../l-bm4Zqaj9bJZ8P9cBqRijpiIabNL1UHwocRI55YIXjA9sXgZB...

But the same message. That is why I asked for an example.

**Comment by tmknight** [comment]
This is the code for the apprise call, is it possible the `|` in the body are problematic: https://github.com/tmknight/docker-autoheal/blob/1e97664da65f21374abf1e2c395877316ae9b511/src/execute/action.rs#L81

**Comment by saschabrockel** [comment]
I don't know, I never used Rust. Is it working for you? Is Apprise now included or does it expect its own local instance with this URL?

**Comment by tmknight** [comment]
It expects you to have a working solution; we just send to it via API.

**Comment by saschabrockel** [comment]
Okay, most solutions include shoutrrr or apprise for this. So I tried it just with webhook as I mentioned before and it's not working. No idea how you expect this to work. If I see your code I'm not smarter than before.

I even installed apprise and added the URL now as apprise URL variable and still `Remote file does not exist -- broken link!!!` is that a joke? I don't even see that line in your code. I tested it with cURL. 

`curl -X POST -d '{"body":"test body","title":"test title"}'     -H "Content-Type: application/json"     "http://192.168.178.29:8004/notify/d0f921cc9ebc55bd9126cbab6f40518a0aee0d1d4ecce7cb41fb7214780c8445"` works

**Comment by tmknight** [comment]
This is not intended to be an all-in-one, throw in the kitchen sink monitoring and alerting solution.  It is meant to monitor and remediate docker containers to the best that docker APIs and docker status messages allow.  Webhook and Apprise are nice-to-haves and work fine in most scenarios.  If you cannot live with those not working, I suggest trying another tool or, better, feel free to dig into the code and suggest how it can be improved.

**Comment by saschabrockel** [comment]
Why so mad? I was just wondering as the original project included a possibility for webhooks. And then I saw that you extended the features like a label to exclude containers which is very useful. I didn't know you rewrote the whole code base. I never wrote any Rust code. I can just tell you that the environment variables you propose for webhooks / apprise do not work. Of course its no alerting solution. I used the original image of willfarrell long enough to know what this tool is doing.
